main.py

```diff
@@ -47,59 +55,75 @@
 
 @app.get("/list_uris")
 async def list_uris() -> JSONResponse:
-    """Coroutine to list all the uris from all the event services
-    
-    Returns:
-        JSONResponse: the list of uris as a json.
-    
-    Usage:
-        curl -X GET "http://localhost:8042/list_uris"
-    """
-    all_uris = {}
-
-    for service_name, client in clients.items():
-        # get the list of uris from the event service
-        uris: list[Uri] = []
-        try:
-            # NOTE: some services may not be available, so we need to handle the timeout
-            uris = await asyncio.wait_for(client.list_uris(), timeout=0.1)
-        except asyncio.TimeoutError:
-            continue
-
-        # convert the uris to a dict, where the key is the uri full path
-        # and the value is the uri proto as a json string
-        for uri in uris:
-            all_uris[f"{service_name}{uri.path}"] = json.loads(MessageToJson(uri))
-
-    return JSONResponse(content=all_uris, status_code=200)
-
+    """Return all the uris from the event manager."""
+    all_uris_list: EventServiceConfigList = event_manager.get_all_uris_config_list(
+        config_name="all_subscription_uris"
+    )
 
-@app.websocket("/subscribe/{service_name}/{uri_path}")
-async def subscribe(websocket: WebSocket, service_name: str, uri_path: str, every_n: int = 1):
+    all_uris = {}
+    for config in all_uris_list.configs:
+        if config.name == "all_subscription_uris":
+            for subscription in config.subscriptions:
+                uri = subscription.uri
+                service_name = uri.query.split("=")[1]
+                key = f"{service_name}{uri.path}"
+                value = {
+                    "scheme": "protobuf",
+                    "authority": config.host,
+                    "path": uri.path,
+                    "query": uri.query,
+                }
+                all_uris[key] = value
+
+    return JSONResponse(content=dict(sorted(all_uris.items())), status_code=200)
+
+
+@app.websocket("/subscribe/{service_name}/{uri_path:path}")
+@app.websocket("/subscribe/{service_name}/{sub_service_name}/{uri_path:path}")
+async def subscribe(
+    websocket: WebSocket,
+    service_name: str,
+    uri_path: str,
+    sub_service_name: str = None,
```

Comment:
```suggestion
    sub_service_name: Optional[str] = None,
```

Reply:
None 's are typed with Optional

Reply:
Good point! Addressed in commit 80ddebdcd28e5406ab03b4feeb258033ebc18f47